include/deal.II/numerics/data_out_dof_data.templates.h

```diff
@@ -1238,26 +1266,27 @@ DataOut_DoFData<DoFHandlerType,patch_dim,patch_space_dim>::get_vector_data_range
           // by the appropriate amount, same
           // for 'i', since we have already
           // dealt with all these components
-          output_component += patch_space_dim-1;
-          i += patch_space_dim-1;
+          output_component += patch_space_dim;
+          i += patch_space_dim;
+        }
+      else
+        {
+          // just move one component forward by one, or two if the vector
+          // happens to be complex-valued
+          if ((*d)->is_complex_valued() == false)
+            {
+              ++output_component;
+              ++i;
+            }
+          else
+            {
+              output_component += 2;
+              ++i;
+            }
         }
 
-  // note that we do not have to traverse the
-  // list of cell data here because cell data
-  // is one value per (logical) cell and
-  // therefore cannot be a vector
-
-  // as a final check, the 'component'
```

Comment:
i presume this is not valid anymore as one can have an arbitrary mixture of complex and real-valued vectors

Reply:
I removed the assertion because it has become fundamentally more difficult to check, and because its correctness is tested in redundant ways both in the testsuite and in other places in the code. It's just become more complicated to calculate the number of output components and adding them add requires a bit more code. But we do this in other places of the code, and so I'm not concerned about losing this one assertion here.